api/src/main/java/org/openmrs/api/impl/ProgramWorkflowServiceImpl.java

```diff
@@ -584,5 +588,48 @@ public ProgramWorkflow getWorkflow(Integer workflowId) {
 	public ProgramWorkflow getWorkflowByUuid(String uuid) {
 		return dao.getWorkflowByUuid(uuid);
 	}
-	
+        
+        @Override
+        public List<ProgramAttributeType> getAllProgramAttributeTypes() {
+            return dao.getAllProgramAttributeTypes();
+        }
+
+        @Override
+        public ProgramAttributeType getProgramAttributeType(Integer id) {
+            return dao.getProgramAttributeType(id);
+        }
+        
+        @Override
+        public ProgramAttributeType getProgramAttributeTypeByUuid(String uuid) {
+            return dao.getProgramAttributeTypeByUuid(uuid);
+        }
+
+        @Override
+        public ProgramAttributeType saveProgramAttributeType(ProgramAttributeType type) {
+            return dao.saveProgramAttributeType(type);
+        }
+
+        @Override
+        public void purgeProgramAttributeType(ProgramAttributeType type) {
+            dao.purgeProgramAttributeType(type);
+        }
+
+        @Override
+        public PatientProgramAttribute getPatientProgramAttributeByUuid(String uuid) {
+            return dao.getPatientProgramAttributeByUuid(uuid);
+        }
+
+        @Override
+        public Collection<Encounter> getEncountersByPatientProgramUuid(String patientProgramUuid) {
+            return Collections.EMPTY_LIST;
```

Comment:
Why return an empty list here?

Reply:
that was for testing purposes only, let me change that.